core/src/main/java/com/datastax/oss/driver/api/core/data/AccessibleByName.java

```diff
@@ -56,4 +56,18 @@
    * @throws IndexOutOfBoundsException if the index is invalid.
    */
   DataType getType(String name);
+
+  /**
+   * Try to return the first index where a given identifier appears.
+   *
+   * @see AccessibleByName#firstIndexOf
+   * @throws IllegalArgumentException if name was not found
+   */
+  default int tryFirstIndexOf(String name) {
+    final int indexOf = firstIndexOf(name);
+    if (indexOf == -1) {
+      throw new IllegalArgumentException(name + " is not a column defined in this metadata");
+    }
+    return indexOf;
+  }
```

Comment:
This looks good to me, but since `firstIndexOf` is not called from anywhere else anymore (actually it is from `DefaultRow.getType`, but we would benefit from the improved message there as well), we might as well modify the contract of `firstIndexOf` directly,  instead of adding a separate method: specify in the javadoc that an `IllegalArgumentException` should be thrown, and change the 4 existing implementations. This duplicates the check in each implementation, but we can also give an even better message: "is not a [column|field|variable] in this [row|UDT|bound statement]".